plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadataFactory.java

```diff
@@ -52,7 +52,7 @@ public IcebergMetadataFactory(
             IcebergFileSystemFactory fileSystemFactory,
             TableStatisticsWriter tableStatisticsWriter,
             @RawHiveMetastoreFactory Optional<HiveMetastoreFactory> metastoreFactory,
-            @ForIcebergScanPlanning ExecutorService executor,
+            @ForIcebergScanPlanning ExecutorService systemTableExecutor,
```

Comment:
This executor is not specific to system tables, it is used for planning any iceberg scan which could be system table or not.
Either we shouldn't rename it or rename to icebergScanExecutor

Reply:
I've renamed it to `icebergScanExecutor` as you suggested. Keeping it as just `executor` could cause confusion, especially since I'm adding a new executor named `metadataFetchingExecutor`.